mne/viz/topomap.py

```diff
@@ -370,37 +370,30 @@ def _griddata(x, y, v, xi, yi):
     d = np.abs(d - d.T)
     n = d.shape[0]
     d.flat[::n + 1] = 1.
-
     g = (d * d) * (np.log(d) - 1.)
     g.flat[::n + 1] = 0.
     weights = linalg.solve(g, v.ravel())
 
-    m, n = xi.shape
-    zi = np.zeros_like(xi)
-    xy = xy.T
-
-    g = np.empty(xy.shape)
-    for i in range(m):
-        for j in range(n):
-            d = np.abs(xi[i, j] + -1j * yi[i, j] - xy)
-            mask = np.where(d == 0)[0]
-            if len(mask):
-                d[mask] = 1.
-            np.log(d, out=g)
-            g -= 1.
-            g *= d * d
-            if len(mask):
-                g[mask] = 0.
-            zi[i, j] = g.dot(weights)
+    d = np.abs(xi + -1j * yi - xy[:, np.newaxis, np.newaxis])
+    mask = (d == 0)
+    d[mask] = 1.
+    g = np.log(d)
+    g -= 1.
+    g *= d * d
+    g[mask] = 0.
+    zi = np.einsum('i,ijk->jk', weights, g)
     return zi
 
 
 def _plot_sensors(pos_x, pos_y, sensors, ax):
     """Plot sensors."""
     from matplotlib.patches import Circle
+    from matplotlib.collections import PatchCollection
     if sensors is True:
+        patches = list()
         for x, y in zip(pos_x, pos_y):
-            ax.add_artist(Circle(xy=(x, y), radius=0.003, color='k'))
+            patches += [Circle(xy=(x, y), radius=0.003)]
+        ax.add_collection(PatchCollection(patches, color='k'))
```

Comment:
I think the scatter option might give another decent boost.
Though most of the test comes from creating (and perhaps closing) the new axes, not the actual plot (it obscures the differences between the actual sensor plotting: I think scatter is actually ~2x as fast as the patches, not just a bit.)
So then it's possible we could save a lot with the axes divider pattern ..?